### DISCUSSION TITLE
SD card content does not update when inserted into a card extender

### DISCUSSION THREAD

**Discussion by laserfan01** [mriscoc/Ender3V2S1#643] [General]
I recently installed the pro firmware for my ender 3v2 with BL touch "ender3v2-422-blubl-mpc-20221222". It seems to work OK until I try to print.  I get a "no media inserted" error.  I went thru all the menus to see if there was a way to enable reading the SD card.  I really do like the firmware and plan to go to the 300C version because I have the all metal hot end w/ direct drive.  I have on order the 300C thermistor from slice engineering.  I have been waiting for firmware like this.  I tried to compile my own but failed.  I tried all three version for the 4.22 board and they all produce the same result.   I am open to suggestions.  This a giant leap in firmware for the enders.  I have now Jyers and it works but I want the higher temps. 
Thanks......... Hank 
![20230112_133806](https://user-images.githubusercontent.com/48221538/212164671-d1b3a5f1-8e4e-4187-95e8-cf0c02039ae8.jpg)

**Comment by mriscoc** [comment]
Do you have a SD memory card extender?

**Reply by laserfan01** [reply]
yes I do but never had a problem with it.

**Reply by mriscoc** [reply]
Go to Advanced Settings menu and active `Media Update`. 

The Card extender does not have an SD card detector switch. It's a mechanical switch which is usually closed if an SD-card is inserted in the connector. It's used to detect the presence of the card, the printer re-mounts the SD file system when a card insertion is detected. In old firmware versions, for compatibility with card extender, the SD media was forced to be re-mounted each time that user enters in Print/Media menu, that behavior is not optimal for people without card extenders, so now, only when the `Media Update` option is activated, the SD file system is re-mounted.

**Reply by laserfan01** [reply]
Thanks again for all the help.  I am now up and printing.

**Reply by spectangulation** [reply]
> Go to Advanced Settings menu and active `Media Update`.
> 
> The Card extender does not have an SD card detector switch. It's a mechanical switch which is usually closed if an SD-card is inserted in the connector. It's used to detect the presence of the card, the printer re-mounts the SD file system when a card insertion is detected. In old firmware versions, for compatibility with card extender, the SD media was forced to be re-mounted each time that user enters in Print/Media menu, that behavior is not optimal for people without card extenders, so now, only when the `Media Update` option is activated, the SD file system is re-mounted.

goated
